bundles/plugins/org.bonitasoft.studio.common.diagram/src/org/bonitasoft/studio/common/gmf/tools/tree/MainProcessFormMappingViewerFilter.java

```diff
@@ -34,20 +34,7 @@ public boolean select(final Viewer viewer, final Object parentElement, final Obj
     }
 
     private boolean selectFormMapping(final FormMapping formMapping) {
-        return formMapping.eContainer() instanceof MainProcess ? false : selectNonEmptyFormMapping(formMapping);
-    }
-
-    private boolean selectNonEmptyFormMapping(final FormMapping formMapping) {
-        switch (formMapping.getType()) {
-            case LEGACY:
-                return false;
-            case URL:
-                return true;
-            case INTERNAL:
-                return formMapping.getTargetForm() != null && formMapping.getTargetForm().hasContent();
-            default:
-                return false;
-        }
+        return !(formMapping.eContainer() instanceof MainProcess);
```

Comment:
we are checking here also if form mapping is not in MainProcess... we might can be in this state quite often :s we shoudl relaly try not be in this state